client/src/controller/UserController.ts

```diff
@@ -0,0 +1,233 @@
+                        <td>${user.lastName}</td>
+                        <td>${user.email}</td>
+                        <td>${user.phone}</td>
+                        <td>${user.address}</td>
+                        <td>${user.role}</td>`;
+    const actionTd = document.createElement("td");
+    row.appendChild(actionTd);
+    const editbutton = this.createNewButton("Edit", CC.editbutton);
+    const deletebutton = this.createNewButton("Delete", CC.deletebutton);
+    const cancelbutton = this.createNewButton("Cancel", CC.cancelbutton);
+    const savebutton = this.createNewButton("Save", CC.savebutton);
+    cancelbutton.style.display = "none";
+    savebutton.style.display = "none";
+
+    editbutton.addEventListener("click", () => {
+      this.edit(index);
+    });
+    savebutton.addEventListener("click", () => {
+      this.save(index);
+    });
+    cancelbutton.addEventListener("click", () => {
+      this.cancel(index);
+    });
+    deletebutton.addEventListener("click", () => {
+      this.delete(index);
+    });
+    actionTd.appendChild(editbutton);
+    actionTd.appendChild(deletebutton);
+    actionTd.appendChild(savebutton);
+    actionTd.appendChild(cancelbutton);
+
+    this.tableBody!.appendChild(row);
+  }
+  createNewButton(text: string, className: CC): HTMLButtonElement {
+    const button = document.createElement("button");
+    button.className = className;
+    button.textContent = text;
+
+    return button;
+  }
+
+  editRow(i: number) {
+    const row = this.tableBody!.children[i] as HTMLTableRowElement;
+    const buttonTd = row.lastChild as HTMLTableDataCellElement;
+
+    (
+      buttonTd.querySelector(`.${CC.editbutton}`) as HTMLButtonElement
+    ).style.display = "none";
+    (
+      buttonTd.querySelector(`.${CC.deletebutton}`) as HTMLButtonElement
+    ).style.display = "none";
+    (
+      buttonTd.querySelector(`.${CC.savebutton}`) as HTMLButtonElement
+    ).style.display = "";
+    (
+      buttonTd.querySelector(`.${CC.cancelbutton}`) as HTMLButtonElement
+    ).style.display = "";
+    const user = this.users[i];
+    row.children[0].innerHTML = `<input value=${user.firstName}>`;
+    row.children[1].innerHTML = `<input value=${user.middleName}>`;
+    row.children[2].innerHTML = `<input value=${user.lastName}>`;
+    row.children[3].innerHTML = `<input value=${user.email}>`;
+    row.children[4].innerHTML = `<input value=${user.phone}>`;
+    const select = document.createElement("select");
+    select.className = "selectrole";
+    for (const e in Role) {
+      const option = document.createElement("option");
+      option.value = e;
+      option.textContent = e;
+      if (user.role === e) option.selected = true;
+      else option.selected = false;
+      select.appendChild(option);
+    }
+    row.children[5].innerHTML = `<input value=${user.address}>`;
+    row.children[6].firstChild!.replaceWith(select);
+  }
+  create(user: User) {
+    this.users.push(user);
```

Comment:
Why is this not `this.userServicesCrud.create(user)`

Reply:
this is extra code, there is no use of it will remove it